### PR TITLE
Merge with updates

### PR DESCRIPTION
**Pull request by rez5427**
issues https://github.com/riscv/sail-riscv/issues/583

### CODE REVIEW

**Comment by rez5427** [comment]
#586

File: model/riscv_sys_regs.sail [hunk 1]
```diff
@@ -279,8 +279,7 @@ function legalize_mstatus(o : Mstatus, v : xlenbits) -> Mstatus = {
 
   /* Hardwired to zero in the absence of 'U' or 'N'. */
   let m = if not(extensionEnabled(Ext_N)) then {
-             let m = [m with UPIE = 0b0];
-             let m = [m with UIE = 0b0];
+             let m = [m with UPIE = 0b0, UIE = 0b0];
```
Comment (Timmmm) [inline model/riscv_sys_regs.sail:282]: Can remove the `let m =` variable.

File: model/riscv_sys_regs.sail [hunk 2]
```diff
@@ -577,37 +576,38 @@ function lower_mstatus(m : Mstatus) -> Sstatus = {
...
+    MXR = s[MXR],
+    SUM = s[SUM],
+    XS = s[XS],
+    FS = s[FS],
+    VS = s[VS],
+    SPP = s[SPP],
+    SPIE = s[SPIE],
+    UPIE = s[UPIE],
+    SIE = s[SIE],
+    UIE = s[UIE]
```
Comment (Timmmm) [inline model/riscv_sys_regs.sail:609]: Can you add a trailing comma here?

File: model/riscv_sys_regs.sail [hunk 3]
```diff
@@ -645,26 +645,29 @@ bitfield Sinterrupts : xlenbits = {
...
-  let s = [s with UTI = m[UTI] & d[UTI]];
-  let s = [s with USI = m[USI] & d[USI]];
-  s
+  [s with
+    SEI = m[SEI] & d[SEI],
+    STI = m[STI] & d[STI],
+    SSI = m[SSI] & d[SSI],
+    UEI = m[UEI] & d[UEI],
+    UTI = m[UTI] & d[UTI],
+    USI = m[USI] & d[USI]
```
Comment (Timmmm) [inline model/riscv_sys_regs.sail:655]: Trailing comma.

File: model/riscv_sys_regs.sail [hunk 4]
```diff
@@ -645,26 +645,29 @@ bitfield Sinterrupts : xlenbits = {
...
-  let s = [s with USI = m[USI] & d[USI]];
-  s
+
+  [s with
+    SEI = m[SEI] & d[SEI],
+    STI = m[STI] & d[STI],
+    SSI = m[SSI] & d[SSI],
+    UEI = m[UEI] & d[UEI],
+    UTI = m[UTI] & d[UTI],
+    USI = m[USI] & d[USI]
```
Comment (Timmmm) [inline model/riscv_sys_regs.sail:669]: And here

**Review by Timmmm — CHANGES_REQUESTED** [review]
Very nice, thanks! Just a couple of minor nits and then we can merge it.

**Review by Timmmm — APPROVED** [review]

**Comment by Timmmm** [comment]
Thanks! I'll merge this tomorrow.

**Review by arichardson — APPROVED** [review]
History is messy but as long as this is squash-merged with a sensible commit message it LGTM.

**Pull request merged**